packages/core/src/domain/error/trackRuntimeError.spec.ts

```diff
@@ -61,196 +54,217 @@ describe('instrumentOnError', () => {
   const testColNo = 42
   const ERROR_MESSAGE = 'foo'
 
-  let onErrorSpy: jasmine.Spy
-  let callbackSpy: jasmine.Spy<UnhandledErrorCallback>
-  let stopCollectingUnhandledError: () => void
-  let originalOnErrorHandler: OnErrorEventHandler
-
-  beforeEach(() => {
-    originalOnErrorHandler = window.onerror
-    onErrorSpy = jasmine.createSpy()
-    window.onerror = onErrorSpy
-    callbackSpy = jasmine.createSpy()
-    ;({ stop: stopCollectingUnhandledError } = instrumentOnError(callbackSpy))
-  })
-
-  afterEach(() => {
-    window.onerror = originalOnErrorHandler
-    stopCollectingUnhandledError()
-  })
+  const withInstrumentOnError = async (callback: () => void): Promise<jasmine.Spy> => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏The name is a bit unclear. To better capture the intention what about: `spyAndInstrumentOnError`?

Reply:
Sure, SGTM! I'll make that change.

Reply:
I renamed all of the test helpers. After trying to find a naming pattern that worked well for all of them I ended up going with `errorViaTrackRuntimeError`, `spyViaInstrumentOnError`, and `spyViaInstrumentOnUnhandledRejection`.